engine/api/workflow/dao_run.go

```
@@ -644,12 +644,14 @@ func CreateRun(db *gorp.DbMap, wf *sdk.Workflow, opts sdk.WorkflowRunPostHandler
 			return nil, err
 		}
 
-		// Add service for consumer if exists
-		s, err := services.LoadByConsumerID(context.Background(), db, c.ID)
-		if err != nil && !sdk.ErrorIs(err, sdk.ErrNotFound) {
-			return nil, err
+		if c.AuthConsumerUser != nil {
```

Maybe this should return an error if AuthConsumer is not set